src/xhr.js

```diff
@@ -82,23 +82,29 @@ const xhrFactory = function() {
 
     // Allow an optional user-specified function to modify the option
     // object before we construct the xhr request
+    // TODO: Remove beforeRequest in the next major release.
     const beforeRequest = XhrFunction.beforeRequest || videojs.Vhs.xhr.beforeRequest;
     // onRequest and onResponse hooks as a Set, at either the player or global level.
-    const _requestCallbackSet = XhrFunction._requestCallbackSet || videojs.Vhs.xhr._requestCallbackSet;
+    // TODO: new Set added here for beforeRequest alias. Remove this when beforeRequest is removed.
+    const _requestCallbackSet = XhrFunction._requestCallbackSet || videojs.Vhs.xhr._requestCallbackSet || new Set();
     const _responseCallbackSet = XhrFunction._responseCallbackSet || videojs.Vhs.xhr._responseCallbackSet;
 
     if (beforeRequest && typeof beforeRequest === 'function') {
-      const newOptions = beforeRequest(options);
-
-      if (newOptions) {
-        options = newOptions;
-      }
+      videojs.log.warn('beforeRequest is deprecated, use onRequest instead.');
+      _requestCallbackSet.add(beforeRequest);
     }
 
     // Use the standard videojs.xhr() method unless `videojs.Vhs.xhr` has been overriden
     // TODO: switch back to videojs.Vhs.xhr.name === 'XhrFunction' when we drop IE11
     const xhrMethod = videojs.Vhs.xhr.original === true ? videojsXHR : videojs.Vhs.xhr;
 
+    // call all registered onRequest hooks
+    callAllHooks(_requestCallbackSet, options);
```

Comment:
It looks like `options` is passed here but no return value is used, which would be a change from the above removed behavior where `beforeRequest` returned a new options object. It seems that this new expectation is that each request callback would reference the same options object - each potentially modifying the same object in sequence.

I don't think this is a bad implementation, but it could be a breaking change to `beforeRequest`. Consider this would likely fail with this refactor:

```js
videojs.Vhs.xhr.beforeRequest = (options) => {
  // Ignores `options` and returns a totally new object.
  return {...};
};
```
We probably don't want that to break, so perhaps the `callAllHooks` should have special handling for the `beforeRequest`/`onRequest` hooks where it passes the return value of the previous hook through a chain and returns the last result?

Reply:
Great catch @misteroneill ! I added some logic to support both cases and a test for the use case you described. Let me know what you think.